stat_fem/ObsData.py

```diff
@@ -0,0 +1,215 @@
+import numpy as np
+from firedrake import COMM_WORLD, COMM_SELF
+from .covariance_functions import sqexp, sqexp_deriv
+
+class ObsData(object):
+    """
+    Class representing Observational Data and discrepancy between model and observations
+
+    This class holds information on observational data and uncertainties. It also serves
+    as a wrapper to compute the model discrepancy (since the data coordinates and
+    uncertainties are stored here). At the moment, it only implements the squared
+    exponential kernel and assumes that data is not time varying.
```

Comment:
Yes, that is correct. The library can't handle time-dependent problems at the moment as the original theory paper doesn't consider this. However, this isn't really a limitation for the long term, as most FEM problems handle time discretization separately from the FEM solution by doing some kind of Runge-Kutta integration and doing an appropriate FEM solve during each RK iteration. So the user would just need to update the data in march-step with the RK integration (though we might eventually want to provide some convenience functions for this). However, estimating parameters for the model discrepancy isn't really supported doing this, so that is one thing that I think does require a bit of thought.

Reply:
Added issue #26 to flag this.